### ISSUE TITLE
A11y_SQLAzureDataStudio_ConnectToASQLServer_AdvancedProperties_ScreenReader: Focus order while navigating the scrollable content is improper due to which all content present is not read by screen reader.

### ISSUE THREAD

**Issue by v-atkulk** [microsoft/azuredatastudio#9204]
“[Check out Accessibility Insights! ](https://accessibilityinsights.io/)- Identify accessibility bugs before check-in and make bug fixing faster and easier.”

#A11yMAS;#A11yTCS;#SQL Azure Data Studio;#DesktopApp;#Win32;#MAS2.4.3; #MAS1.3.1;#NVDA;#Benchmark;#A11y_SQLAzureDataStudioBenchmark;#A11ySev2;#JAWS;

**Environment Details:**
Version: 1.16.0-insider
Commit: 506c6a5e5f8423d3249670f55b1466b1980a5f03
Date: 2020-02-17T05:43:18.399Z
VS Code: 1.42.0
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363

**Additional Details:**
Screen reader: NVDA(2019.3.1), JAWS(2020.2001.70)
MAS reference: MAS2.4.3, MAS1.3.1

**Repro Steps:**

1. Launch Azure Data Studio.
2. Select the Connections icon (if not already selected) from left bar.
3. Hover over the 'Servers' header and select the 'New Connection' icon.
4. Navigate till Advanced button under Connection details & hit enter.
5. Launch JAWS screen reader.
6. Check if all the data present on the screen is been read by screen reader.

**Actual:**
When we navigate through the information that is present on advanced properties screen, the content which is visible when we scroll the content is not been read by JAWS screen reader.

**Same issue observed with NVDA also.**

**Expected:**
All the content that is present on the screen should be read by the screen reader.

**User Impact:**
Users might receive less information as all the content present on the screen is not accessible for keyboard as well as screen reader users.

**Attachment link reference:**
[9204_A11y_SQLAzureDataStudio_ConnectToASQLServer_AdvancedProperties_ScreenReader.zip](https://github.com/microsoft/azuredatastudio/files/4291556/9204_A11y_SQLAzureDataStudio_ConnectToASQLServer_AdvancedProperties_ScreenReader.zip)

**Comment by alanrenmsft** [comment]
@v-atkulk please fix the attachment, i am not able to open it.

i am assuming you are saying not all controls are accessible using keyboard

**Comment by Charles-Gagnon** [comment]
So there's definitely a bug here with the tab navigation only taking into account visible items and so the items outside the current view aren't tab-navigable. This is probably due to my changes a while back to fix this for a separate issue - https://github.com/microsoft/azuredatastudio/pull/8326

I'm investigating currently

**Comment by Charles-Gagnon** [comment]
Hm, this is an interesting one. So I don't think my changes actually caused this - the reason this is happening is because the items are being completely removed from the DOM as they scroll out of view. This is done by the base split view class and isn't something we have control over :

https://github.com/microsoft/azuredatastudio/blob/master/src/sql/base/browser/ui/scrollableSplitview/scrollableSplitview.ts#L691

The changes I made should work correctly as long as the items are in the dom - but since they aren't it just does its best. 

A couple options here :

1. Try to figure out why VS Code is doing this and whether we can disable this behavior here
1. Use a different view container. For example I think it's probably reasonable here to just use a normal list view with sections that aren't collapsible. 
1. Try to refactor the logic I added to take this behavior into account. 

I'm leaning towards 2 for the sake of simplicity. @alanrenmsft any thoughts here?

**Comment by swethakodigi** [comment]
[9204_A11y_SQLAzureDataStudio_ConnectToASQLServer_AdvancedProperties_ScreenReader.zip](https://github.com/microsoft/azuredatastudio/files/10765099/9204_A11y_SQLAzureDataStudio_ConnectToASQLServer_AdvancedProperties_ScreenReader.zip)

verified the issue in the below it is not repro'ing. Hence closing this bug
Version: 1.41.1 (user setup)
Commit: 33dce95b152227c4f10f6251ab30690fe9895c0c
Date: 2023-01-28T00:46:39.274Z
VS Code: 1.67.0
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.22621

**Comment by swethakodigi** [comment]
GitHubTags:#Closed

**Comment by Ashutoshsahoo164** [comment]
GitHubTags:#Closed;

**Comment by mstechie** [comment]
GitHubTags:#SQLADS-Win32-Feb2020;
